api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/N3Template.java

```diff
@@ -34,13 +52,89 @@ public Parameters getProvidedParams() {
 		return new Parameters();
 	}
 
+	public String getN3Text() { return this.n3Text; }
+
 	//endregion
 
 	@Override
 	public OperationResult run(OperationData input) {
-		return null;
+		if (!isInputValid(input)) {
+			return new OperationResult(500);
+		}
+
+		String substitutedN3Template;
+		try {
+			substitutedN3Template=insertParameters(input);
+		}catch (InputMismatchException e){
+			log.error(e);
+			return new OperationResult(500);
+		}
+
+		List<Model> additionModels;
+		try {
+			additionModels = ProcessRdfForm.parseN3ToRDF(Arrays.asList(substitutedN3Template), ProcessRdfForm.N3ParseType.REQUIRED);
+		} catch (Exception e) {
+			log.error("Error while trying to parse N3Template string and create a Jena rdf Model");
+			log.error(e);
+			return new OperationResult(500);
+		}
+
+		AdditionsAndRetractions changes = new AdditionsAndRetractions(additionModels, new ArrayList<Model>());
+		Model writeModel = ModelAccess.on(input.getContext()).getOntModel(templateModel.getName());
+		ProcessRdfForm.applyChangesToWriteModel(changes, null, writeModel,"");
+
+		return new OperationResult(200);
+	}
+
+	private String insertParameters(OperationData input) throws InputMismatchException{
+
+		EditN3GeneratorVTwo gen = new EditN3GeneratorVTwo();
+
+		//Had to convert String to List<String> to be compatible with EditN3GeneratorVTwo methods
+		List<String> n3WithParameters = Arrays.asList(n3Text);
+
+		//region Substitute IRI variables
+		Map<String, List<String>> parametersToUris = requiredParams.params.values().stream()
+				.filter(value->value.getType().getName()=="anyURI")
```

Comment:
What is "anyURI"? Maybe create a constant for it?

Reply:
XSD Datatype for Uniform Resource Identifier (http://books.xmlschemata.org/relaxng/ch19-77009.html). Ok, I will create anyURI constant later today.